### ISSUE TITLE
Block Models Do Not Render In Inventory

### ISSUE THREAD

**Issue by Kalman98** [TEAMModding/KalStuff#11]
Both the chicken nest and trash can models do not render properly in the player's inventory. They show up as only 2D views from the side of the models. I have been trying to fix this for a while and still have not managed it, and so I am posting it as an issue here. I will continue to work on finding a solution, but I feel like I am missing something pretty obvious. There has to be something simple we need to add to the JSON file, I just do not know what that is.

**Comment by pianoman373** [comment]
I'm gonna let you guys solve this one yourselves. The problem is in the Json file. The format changed in 1.9.

Message.sender = "Name removed";

**Comment by Kalman98** [comment]
So basically you don't know how to fix it.

**Comment by ParkerMc** [comment]
I would but I do not have time I have work and robot camp here.

**Comment by pianoman373** [comment]
Don't know how to fix it huh?.....
![2016-07-12_16 57 29](https://cloud.githubusercontent.com/assets/11893049/16785112/f375e494-4851-11e6-8983-3bddcc595aeb.png)

It's a simple fix, it just requires some digging through minecraft's json files to figure out what to add. I could just push it but i figure you guys will learn more by finding it yourselves.

Some hints would be looking through the walking stick code and seeing that translations and rotations
changed to be per-hand.

You could simply add those translations and rotations to the trash can, but there must be an easier way.

Try making the trash can item model the child of another simple block, it will end up working fine.

So something is included in simpler blocks that does the translations and rotations, but the trash can model doesn't have it......

Do enough digging through minecraft's json  files and you'll see the problem.

**Comment by Kalman98** [comment]
Okay, I'm not trying to be rude, but that is just very bad thinking holding back the development of the mod because you think we will learn more by figuring it out ourselves. I have dug through a gazillion JSON files, from Vanilla Minecraft to some of the most famous mods, I have searched the Minecraft Forge forums and the Internet along with some of the most famous modder's tutorial/blogging websites and still could not come up with an answer. As a matter of fact I did even copy the walking stick JSON and use it for the chicken nest and it still did not work. I have finally located the block class however and fixed the problem due to your answer. I do not know how I still had not found the class before, but I have now.

In summary, the bug is fixed and I have not learned any more than I would have if you had directly told me what had changed, although I am now somewhat annoyed that I spent over three hours trying to solve something which you were deliberately not giving me the answer to.

**Comment by pianoman373** [comment]
I'm sorry you had to spend so much time trying to find the problem. I just got home about an hour ago and then I looked through for a way to solve the bug. Once I found it, I thought it would be best to give hints rather than just say "adding the block/block code fixes the problem" I figured looking through the json files and actually seeing the translations in block.json would teach a lot more.

Plus I thought it would be a nice puzzle. I didn't think it would take more than 15 minutes to figure out using the hints. I also wanted to add a simple challenge for circuit, pie, and parker since they haven't contributed in awhile.

**Comment by CircuitLord** [comment]
I don't have my modding workspace setup. I'm pretty busy with life right
now because I'm teaching a robot class every day this week.

**Comment by Kalman98** [comment]
It's okay. It would have been nice if you had said "let's let Circuit, Pie, and Parker handle this one", though. Either way it's fixed now so it doesn't matter anymore.

**Comment by CircuitLord** [comment]
Who is parked?

string MessageSender = "Jordan";
string FinalMessage = "This message was sent by ";
FinalMessage += (MessageSender);
MessageBox.Show(FinalMessage);

**Comment by Kalman98** [comment]
This is getting way off topic now. If you check the GitHub page for this issue, you will see that I updated my comment immediately after sending it and changed that word to "Parker".

**Comment by ParkerMc** [comment]
Hi, 

I've seen some  excellent stuff in the  last catalogue,  just take  a look  at it [LINK REMOVED]

Wishes,

**Comment by ParkerMc** [comment]
Hello! Just take a look at the info I just  found, I'm so excited, check it out at <http://hipriltopha.objectgraph.net/e3tjo>

Pardon my monkey thumbs, nick

**Comment by ParkerMc** [comment]
Dear friend! 

I thought you might be interested it that new  and  very  nice stuff, just take a  look here <http://slekifrona.vantage911.org/e3lll>

nick

**Comment by ParkerMc** [comment]
Greetings! 


I came across that interesting stuff  and I  guess it's something really crazy, just take a look <http://mispaquisle.waterfrontinnorthernmichigan.com/e3cehxc>

Warmest regards, nick

**Comment by ParkerMc** [comment]
Hello, 


This is just something really astonishing, I'm telling you, you have to take  a look <http://frotrothucre.foncieremontblanc.fr/e3ebr>


Sent from a prehistoric stone tablet, nick

**Comment by ParkerMc** [comment]
Hello! 

I just  wanted to say Hello and to show you that stuff, just check  it out <http://paphyngysho.vantage911.us/e3qalt>


nick
